src/Microsoft.Azure.WebJobs/ConnectionProviderAttribute.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// For Attributes that implement <see cref="IConnectionProvider"/> this attribute can be
+    /// applied to the Attribute to specify the attribute Type used for hierarchical
+    /// declarative overrides.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class ConnectionProviderAttribute : Attribute
```

Comment:
My plan is to leave this public. I don't have any reservations doing that - it's already a public pattern in that ServiceBus is doing it. Rather than hack some internal thing, I scoped it to something that I think makes general sense and could be cleanly used by other extensions if they wanted to. I.e. we have (and could surface) attribute resolution methods that know how to "fold" these properly.